src/topp/DatabaseSuitability.cpp

```diff
@@ -0,0 +1,367 @@
+    setValidFormats_("in_novo", { "idXML" });
+    registerOutputFile_("out", "<file>", "", "Optional tsv output", false); //tsv output
+    setValidFormats_("out", { "tsv" });
+    registerIntOption_("novor_fract", "<integer>", 1, "Set the percentage of de novo peptides to capture with a score higher than the fasta score.", false, true);
+    registerFlag_("force_no_re_rank", "Use this flag if you want to disable re-ranking. This might yeild in underperformance.", true);
+  }
+
+
+  // the main_ function is called after all parameters are read
+  ExitCodes main_(int, const char**)
+  {
+    //-------------------------------------------------------------
+    // parsing parameters
+    //-------------------------------------------------------------
+    String in_id = getStringOption_("in_id");
+    String in_spec = getStringOption_("in_spec");
+    String in_novo = getStringOption_("in_novo");
+    String out = getStringOption_("out");
+    Int novo_fract = getIntOption_("novor_fract");
+    bool no_re_rank = getFlag_("force_no_re_rank");
+
+    //-------------------------------------------------------------
+    // reading input
+    //-------------------------------------------------------------
+    IdXMLFile x;
+    vector<ProteinIdentification> prot_ids;
+    vector<PeptideIdentification> pep_ids;
+    x.load(in_id, prot_ids, pep_ids);
+
+    vector<ProteinIdentification> novo_prots;
+    vector<PeptideIdentification> novo_peps;
+    x.load(in_novo, novo_prots, novo_peps);
+
+    UInt64 count_ms2_lvl;
+    {
+      MzMLFile m;
+      PeakFileOptions op;
+      op.addMSLevel(2);
+      op.setFillData(false);
+      m.setOptions(op);
+      PeakMap exp;
+      m.load(in_spec, exp);
+      count_ms2_lvl = exp.size();
+    }
+
+    //-------------------------------------------------------------
+    // calculations
+    //-------------------------------------------------------------
+
+    // db suitability
+
+
+    double cut_off;
+    if (!no_re_rank)
+    {
+      cut_off = getDecoyCutOff_(pep_ids, novo_fract);
+      if (cut_off < 0)
+      {
+        OPENMS_LOG_ERROR << "Could not compute decoy cut off. Re-ranking impossible. If you want to ignore this, set the 'force_no_re_rank' flag." << endl;
+        return INCOMPATIBLE_INPUT_DATA;
+      }
+    }
+
+    UInt64 count_db = 0;
+    UInt64 count_novo = 0;
+    UInt64 count_re_ranked = 0;
+    UInt64 count_interest = 0;
+
+    for (const auto& pep_id : pep_ids)
+    {
+      vector<PeptideHit> hits = pep_id.getHits();
+
+      if (hits.empty()) continue;
+
+      PeptideHit top_hit = hits[0];
+
+      if (top_hit.getMetaValue("target_decoy") == "decoy") continue;
+
+      // check if top hit is found in de novo protein
+      set<String> accessions = top_hit.extractProteinAccessionsSet();
+      bool is_novo = true;
+      for (const String& acc : accessions)
+      {
+        if (acc.find(Constants::UserParam::CONCAT_PEPTIDE) == String::npos)
```

Comment:
technically, the hit could be shared between DB and deNovo...
And should then be counted as DB, right?

Reply:
yes and this should happen here right?
`is_novo` will only stay 'true' as long as each accessions contains `CONCAT_PEPTIDE` and if this isn't found in at least one accession the hit will be counted as DB

Reply:
ahh... right. my bad.

Reply:
Another thing to consider: peptides can (rarely) map to both target and decoy proteins yielding "target+decoy" (I think) for the meta-value. How would you handle those? Most of the time handling them as targets is fine but you should keep it in mind.